### PR TITLE
Create Blog “set-up-apache-skywalking-for-k8s-and-vm-monitoring-in-hpe-greenlake-private-cloud”

### PR DESCRIPTION
**Pull request by GuopingJia**
Automatically generated by Netlify CMS

### CODE REVIEW

File: content/blog/set-up-apache-skywalking-for-k8s-and-vm-monitoring-in-hpe-greenlake-private-cloud.md [hunk 1]
```diff
@@ -0,0 +1,324 @@
...
+    threshold: 8000
+    # The length of time to evaluate the metrics
+    period: 10
+    # How many times after the metrics match the condition, will trigger alarm
+    count: 2
+    # How many times of checks, the alarm keeps silence after alarm triggered, default as same as period.
+    silence-period: 3
+    message: Successful rate of service {name} is lower than 80% in 2 minutes of last 10 minutes
+```
+W﻿e can go to the frontend app by clicking the service `agent::app` from SkyWalking UI Service page. From the below service `agent::app` overview page, it shows **Success Rate 66.66%**.
```
Comment (choukden) [inline content/blog/set-up-apache-skywalking-for-k8s-and-vm-monitoring-in-hpe-greenlake-private-cloud.md:312]: You can go to the....

**Review by choukden — CHANGES_REQUESTED** [review]
Hi Guoping,

Thank you very much for this blog post of your great series.

Here are some suggestions for edits.

Dale will go through the editorial review in the coming day or so, So expect to get additional comments and suggestions for edit in the coming days.

best regards,
Denis

File: content/blog/set-up-apache-skywalking-for-k8s-and-vm-monitoring-in-hpe-greenlake-private-cloud.md [hunk 2]
```diff
@@ -0,0 +1,324 @@
...
+
+We may check further the service's trace page and try to figure out the root cause for this issue.
+
+![](/img/sw-svc-app-trace.png) 
+
+## Conclusion
+
+This blog post took the Apache SkyWalking as the application performance monitoring (APM) tool and showed the detailed process to set it up, as a *self-managed* environment in HPE GreenLake for Private Cloud Enterprise, for monitoring and alerting applications. Using the instrumentation of multiple supported agents from Apache SkyWalking, the application workloads can be easily monitored through the integrated Apache SkyWalking UI, with nice application topology map, tracing details and real-time alarms for any application performance issues.  
+
+I﻿n the next blog, we will show you how to use Apache SkyWalking for Kubernetes monitoring in HPE GreenLake for Private Cloud Enterprise.
```
[the remark this answers is not in the document]
Comment (choukden) [reply content/blog/set-up-apache-skywalking-for-k8s-and-vm-monitoring-in-hpe-greenlake-private-cloud.md:324]: @GuopingJia 

Further to our discussion today and my better understanding of the 2 blog post series, I would suggest the following edit for the last sentence of the Conclusion:

[In the next blog post of the series](https://developer.hpe.com/blog/set-up-apache-skywalking-for-k8s-monitoring-in-hpe-greenlake-for-private-cloud-enterprise/), I will show you how to use Apache SkyWalking APM tool for monitoring of Kubernetes clusters provisioned on HPE GreenLake for Private Cloud Enterprise.

**Review by choukden — CHANGES_REQUESTED** [review]
Hi Guoping,

Further to our discussion today and my better understanding of the 2 blog post series, I would suggest the following edit for the last sentence of the Conclusion:

[In the next blog post of the series](https://developer.hpe.com/blog/set-up-apache-skywalking-for-k8s-monitoring-in-hpe-greenlake-for-private-cloud-enterprise/), I will show you how to use Apache SkyWalking APM tool for monitoring of Kubernetes clusters provisioned on HPE GreenLake for Private Cloud Enterprise.

File: content/blog/set-up-apache-skywalking-for-k8s-and-vm-monitoring-in-hpe-greenlake-private-cloud.md [hunk 3]
```diff
@@ -0,0 +1,324 @@
...
+--set oap.storageType=elasticsearch \
+--set ui.image.tag=9.2.0 \
+--set elasticsearch.imageTag=7.17.1 \
+--set elasticsearch.persistence.enabled=true \
+--set elasticsearch.sysctlInitContainer.enabled=false
+```
+
+After running above commands, t﻿he Apache SkyWalking is installed to the Kubernetes cluster's namespace *skywalking*. It creates the *elasticsearch* as the `StatefulSet`, running a pod on each worker node. It runs the Apache SkyWalking OAP with replicas as 2 to provide high availability. 
+
+It should be noted that the last option *elasticsearch.sysctlInitContainer.enabled=false* is necessary. Otherwise, it will try to set up *vm.max_map_count* using a privileged container during *elasticsearch* installation. It will violate the existing *PodSecurityPolicy* deployed in the HPE GreenLake environment. It therefore will fail the Helm install.
```
[the remark this answers is not in the document]
Comment (GuopingJia) [reply content/blog/set-up-apache-skywalking-for-k8s-and-vm-monitoring-in-hpe-greenlake-private-cloud.md:60]: Yes, it's better to say HPE GreenLake for Private Cloud Enterprise environment.

File: content/blog/set-up-apache-skywalking-for-k8s-and-vm-monitoring-in-hpe-greenlake-private-cloud.md [hunk 4]
```diff
@@ -0,0 +1,324 @@
...
+Events:                   <none>
+```
+
+T﻿he SkyWalking UI can then be accessed in the browser by typing the address *gl2-caas.gl-hpe.local:10037*: 
+
+![](/img/sw-ui.png)
+
+### Deploy a Sample SpringBoot Application
+
+A﻿s the first demo application, create a *SpingBoot* Web app that provides a REST endpoint **/message** to print some nice message. Then generate a *jar* file *springboot-k8s-demo.jar* to the target folder:
```
[the remark this answers is not in the document]
Comment (GuopingJia) [reply content/blog/set-up-apache-skywalking-for-k8s-and-vm-monitoring-in-hpe-greenlake-private-cloud.md:132]: No problem. I've added the mvn build process with the pom.xml to generate this jar package.

File: content/blog/set-up-apache-skywalking-for-k8s-and-vm-monitoring-in-hpe-greenlake-private-cloud.md [hunk 5]
```diff
@@ -0,0 +1,324 @@
...
+Apache SkyWalking provides an alerting mechanism to measure application performance according to a list of pre-defined metrics, e.g., _service_resp_time_, _database_access_resp_time_, and _service_sla_. It will trigger alerting when some metrics reach its pre-defined thresholds. We can define new metrics or customize the existing metrics with new thresholds. 
+
+Here is the alarms page from SkyWalking UI showing all the triggered alerting for deployed multi-tier *music* app: 
+
+![](/img/sw-app-alarms.png)
+
+T﻿he alarms page shows *Successful rate of service agent::app is lower than 80% in 2 minutes of last 10 minutes*. It indicates issue from the frontend app server in multi-tier *music* application.
+
+The alerts are triggered by the following metric alerting rule _service_sla_: 
+
```
[the remark this answers is not in the document]
Comment (GuopingJia) [reply content/blog/set-up-apache-skywalking-for-k8s-and-vm-monitoring-in-hpe-greenlake-private-cloud.md:354]: Added the process to read the content of the alerting rules.

**Comment by GuopingJia** [comment]
Hi, Denis, 

The new patch submitted should address all your comments.

Guoping

File: content/blog/set-up-apache-skywalking-for-k8s-and-vm-monitoring-in-hpe-greenlake-private-cloud.md [hunk 6]
```diff
@@ -0,0 +1,382 @@
...
+Apache SkyWalking is lightweight and scalable. It can be easily set up as a *self-managed* APM tool within an on-premises data center. This avoids leasing customer data to third party services and matches well with the strict security restriction in HPE GreenLake for Private Cloud Enterprise environment. 
+
+## Set up Apache SkyWalking for Application Monitoring
+
+I will take the approach to setting up the Apache SkyWalking as a *self-managed* APM tool within the Kubernetes cluster created in HPE GreenLake for Private Cloud Enterprise. This mainly takes into account the security concerns in HPE GreenLake product environment. 
+
+### Prerequisites
+
+Before we start, make sure we meet the following requirements:
```
Comment (drensing) [inline content/blog/set-up-apache-skywalking-for-k8s-and-vm-monitoring-in-hpe-greenlake-private-cloud.md:33]: Recommend rephrasing to "Before starting, make sure you have the following requirements:"

File: content/blog/set-up-apache-skywalking-for-k8s-and-vm-monitoring-in-hpe-greenlake-private-cloud.md [hunk 7]
```diff
@@ -0,0 +1,382 @@
...
+
+* A Kubernetes cluster, being provisioned in HPE GreenLake for Private Cloud Enterprise;
```
Comment (drensing) [inline content/blog/set-up-apache-skywalking-for-k8s-and-vm-monitoring-in-hpe-greenlake-private-cloud.md:35]: Take the punctuation (the semi colons and period) off of the end of each of the following 3 bullets.

File: content/blog/set-up-apache-skywalking-for-k8s-and-vm-monitoring-in-hpe-greenlake-private-cloud.md [hunk 8]
```diff
@@ -0,0 +1,382 @@
...
+$ helm install skywalking skywalking –n skywalking \
+--set oap.image.tag=9.2.0 \
+--set oap.storageType=elasticsearch \
+--set ui.image.tag=9.2.0 \
+--set elasticsearch.imageTag=7.17.1 \
+--set elasticsearch.persistence.enabled=true \
+--set elasticsearch.sysctlInitContainer.enabled=false
+```
+
+After running above commands, t﻿he Apache SkyWalking is installed to the Kubernetes cluster's namespace *skywalking*. It creates the *elasticsearch* as the `StatefulSet` object, running a pod on each worker node. It runs the Apache SkyWalking Observability Analysis Platform (OAP) with replicas as 2 to provide high availability.
```
Comment (drensing) [inline content/blog/set-up-apache-skywalking-for-k8s-and-vm-monitoring-in-hpe-greenlake-private-cloud.md:58]: Add "the" prior to "above commands" , take "the" out prior to "Apache Skywalking", and change "to the Kubernetes" to "on the Kubernetes" so it reads:

"After running the above commands, Apache Skywalking is installed on the Kubernetes..."

It's a little unclear as to what "It" refers to in the next two sentences. I believe that in sentence #2, "It" refers to the Apache SkyWalking app that was just installed. But I'm not sure that's the case in the third sentence. I'm thinking that "It" here refers more to the system you have just installed it on. If that's the case, it would be better to spell out the complete subject at the beginning of the 3rd sentence rather than using the pronoun "It".

File: content/blog/set-up-apache-skywalking-for-k8s-and-vm-monitoring-in-hpe-greenlake-private-cloud.md [hunk 9]
```diff
@@ -0,0 +1,382 @@
...
+--set oap.storageType=elasticsearch \
+--set ui.image.tag=9.2.0 \
+--set elasticsearch.imageTag=7.17.1 \
+--set elasticsearch.persistence.enabled=true \
+--set elasticsearch.sysctlInitContainer.enabled=false
+```
+
+After running above commands, t﻿he Apache SkyWalking is installed to the Kubernetes cluster's namespace *skywalking*. It creates the *elasticsearch* as the `StatefulSet` object, running a pod on each worker node. It runs the Apache SkyWalking Observability Analysis Platform (OAP) with replicas as 2 to provide high availability. 
+
+It should be noted that the last option *elasticsearch.sysctlInitContainer.enabled=false* is necessary. Otherwise, it will try to set up *vm.max_map_count* using a privileged container during *elasticsearch* installation. It will violate the existing *PodSecurityPolicy* deployed in the HPE GreenLake for Private Cloud Enterprise environment. It therefore will fail the Helm install.
```
Comment (drensing) [inline content/blog/set-up-apache-skywalking-for-k8s-and-vm-monitoring-in-hpe-greenlake-private-cloud.md:60]: In this first sentence, do you really mean "option" or should that read "command"?
Be careful to be clear what the pronouns "it" refer to. The first one (at the beginning of the sentence) is fine. But what is "it" that will try to set up *vm.max_map..."? And in the next sentence, it may be more clear to write "By doing so, it will violate the existing.... Enterprise environment and will therefore fail the Helm install."

File: content/blog/set-up-apache-skywalking-for-k8s-and-vm-monitoring-in-hpe-greenlake-private-cloud.md [hunk 10]
```diff
@@ -0,0 +1,382 @@
...
+replicaset.apps/skywalking-ui-549dc5989f    1         1         1       8m9s
+
+NAME                                    READY   AGE
+statefulset.apps/elasticsearch-master   3/3     8m5s
+
+NAME                           COMPLETIONS   DURATION   AGE
+job.batch/skywalking-es-init   1/1           7m27s      8m6s
+```
+
+You can edit the deployed SkyWalking UI service *skywalking_ui* and change its type from *ClusterIP* to *NodePort*. The service will be automatically mapped to gateway host with an assigned port.
```
Comment (drensing) [inline content/blog/set-up-apache-skywalking-for-k8s-and-vm-monitoring-in-hpe-greenlake-private-cloud.md:96]: Add the word "a" prior to "gateway host" at the end of the second sentence.

File: content/blog/set-up-apache-skywalking-for-k8s-and-vm-monitoring-in-hpe-greenlake-private-cloud.md [hunk 11]
```diff
@@ -0,0 +1,382 @@
...
+Events:                   <none>
+```
+
+As shown in the ***Annotations*** section of the service description above, the SkyWalking UI can then be accessed in the browser by typing the address *gl2-caas.gl-hpe.local:10037*: 
+
+![](/img/sw-ui.png)
+
+### Deploy a sample application: *SpringBoot*
+
+A﻿s the first demo application, create a *SpingBoot* Web app that provides a REST endpoint **/message** to print some nice message. Then generate a *jar* package using the [Apache Maven](https://maven.apache.org/what-is-maven.html) command *mvn* with the *pom.xml* file:
```
Comment (drensing) [inline content/blog/set-up-apache-skywalking-for-k8s-and-vm-monitoring-in-hpe-greenlake-private-cloud.md:132]: Do you really mean "As the first demo application..." or "As **in** the first demo application..." If it's the second, you should rephrase it to "As I had you do in the application I demo'd in my first blog post in this series, create...."

Add a comma after "Then" at the beginning of the next sentence.

File: content/blog/set-up-apache-skywalking-for-k8s-and-vm-monitoring-in-hpe-greenlake-private-cloud.md [hunk 12]
```diff
@@ -0,0 +1,382 @@
...
+
+</project>
+
+$﻿ mvn compile
+$﻿ mvn package
+$ ls target/springboot-k8s-demo.jar
+target/springboot-k8s-demo.jar
+```
+
+You can see the *jar* package *springboot-k8s-demo.jar* is built to the *target* folder. B﻿y building a *Docker* image using this generated *jar* file, the *SpringBoot* Web app can be easily deployed as a containerized application to the Kubernetes cluster.
```
Comment (drensing) [inline content/blog/set-up-apache-skywalking-for-k8s-and-vm-monitoring-in-hpe-greenlake-private-cloud.md:196]: I think it would read more clearly if you change "built to the target folder" to "created in the target folder." If you don't care for that, at least change "to" to "in". Similarly, "to the Kubernetes cluster." should be "in the Kubernetes cluster."

File: content/blog/set-up-apache-skywalking-for-k8s-and-vm-monitoring-in-hpe-greenlake-private-cloud.md [hunk 13]
```diff
@@ -0,0 +1,382 @@
...
+
+Apache SkyWalking provides a list of agents for instrumenting applications. A specific agent per programming language can be used for building into corresponding service which collects application data and exports them to the SkyWalking OAP server.
```
Comment (drensing) [inline content/blog/set-up-apache-skywalking-for-k8s-and-vm-monitoring-in-hpe-greenlake-private-cloud.md:198]: "building into corresponding service" should change to "to build a corresponding service". Also "which" should be changed to "that collects application..."

File: content/blog/set-up-apache-skywalking-for-k8s-and-vm-monitoring-in-hpe-greenlake-private-cloud.md [hunk 14]
```diff
@@ -0,0 +1,382 @@
...
+
+Apache SkyWalking provides a list of agents for instrumenting applications. A specific agent per programming language can be used for building into corresponding service which collects application data and exports them to the SkyWalking OAP server.   
+
+![](/img/sw-agents.png)
+
+I﻿n order to monitor the sample  *SpringBoot* Web app from Apache SkyWalking, we need to download the Java agent and rebuild the image.
+
+![](/img/java-agent.png)
+
+H﻿ere is the *Dockerfile* for building the image with Java agent:
```
Comment (drensing) [inline content/blog/set-up-apache-skywalking-for-k8s-and-vm-monitoring-in-hpe-greenlake-private-cloud.md:206]: "with Java agent:" should either read "with the Java agent:" or "using a Java agent:"

File: content/blog/set-up-apache-skywalking-for-k8s-and-vm-monitoring-in-hpe-greenlake-private-cloud.md [hunk 15]
```diff
@@ -0,0 +1,382 @@
...
+# copy the app jar file
+EXPOSE 8080
+ADD target/springboot-k8s-demo.jar /app/springboot-k8s-demo.jar
+WORKDIR /app
+ENTRYPOINT ["java","-javaagent:/opt/agent/skywalking-agent.jar=agent.namespace=default,agent.service_name=springboot-k8s-app,collector.backend_service=skywalking-oap.skywalking.svc.cluster.local:11800, plugin.jdbc.trace_sql_parameters = true,profile.active=true","-jar","/app/springboot-k8s-app.jar"]
+```
+
+### Monitor SpringBoot application from SkyWalking UI
+
+After build the Docker image *guopingjia/springboot-k8s-demo:pce* and push it to the *DockerHub* registry, we deploy the *SpringBoot* Web app to the Kubernetes cluster with the *deployment.yaml* manifest file:
```
Comment (drensing) [inline content/blog/set-up-apache-skywalking-for-k8s-and-vm-monitoring-in-hpe-greenlake-private-cloud.md:222]: Modify first sentence to "After **building** the Docker image *guopingjia/springboot-k8s-demo:pce* and **pushing** it to the *DockerHub* registry, **deploy** the *SpringBoot* **web** app to the Kubernetes cluster with the *deployment.yaml* manifest file:

File: content/blog/set-up-apache-skywalking-for-k8s-and-vm-monitoring-in-hpe-greenlake-private-cloud.md [hunk 16]
```diff
@@ -0,0 +1,382 @@
...
+
+You may check further the service's trace page and try to figure out the root cause for this issue.
+
+![](/img/sw-svc-app-trace.png)
+
+## Conclusion
+
+This blog post took the Apache SkyWalking as the application performance monitoring (APM) tool and showed the detailed process to set it up, as a *self-managed* environment in HPE GreenLake for Private Cloud Enterprise, for monitoring and alerting applications. Using the instrumentation of multiple supported agents from Apache SkyWalking, the application workloads can be easily monitored through the integrated Apache SkyWalking UI, with nice application topology map, tracing details and real-time alarms for any application performance issues.  
+
+I﻿n [the next blog post of the series](https://developer.hpe.com/blog/set-up-apache-skywalking-for-k8s-monitoring-in-hpe-greenlake-for-private-cloud-enterprise/) , I will show you how to use Apache SkyWalking APM tool for monitoring of Kubernetes clusters provisioned on HPE GreenLake for Private Cloud Enterprise.
```
Comment (drensing) [inline content/blog/set-up-apache-skywalking-for-k8s-and-vm-monitoring-in-hpe-greenlake-private-cloud.md:382]: In this case, you would add the word "the" prior to "Apache SkyWalking APM tool for monitoring...."

**Review by drensing — COMMENTED** [review]
Thank you, Guoping! After this one, just one more for me to review.

All the best,
Dale

File: content/blog/set-up-apache-skywalking-for-k8s-and-vm-monitoring-in-hpe-greenlake-private-cloud.md [hunk 17]
```diff
@@ -0,0 +1,382 @@
...
+
+[Apache SkyWalking](https://skywalking.apache.org/) is an open source APM tool with capabilities for monitoring, tracing and diagnosing distributed system. It’s especially designed for microservices, cloud native and container-based architectures. 
+
+Apache SkyWalking provides a list of agents to be used for building *Java*, *.NET Core*, *PHP*, *NodeJS*, *Golang*, *LUA*, *Rust* and *C++* apps. It provides tracing, metrics analysis, alerting, service mesh observability and visualization.
+
+Apache SkyWalking is lightweight and scalable. It can be easily set up as a *self-managed* APM tool within an on-premises data center. This avoids leasing customer data to third party services and matches well with the strict security restriction in HPE GreenLake for Private Cloud Enterprise environment. 
+
+## Set up Apache SkyWalking for Application Monitoring
+
+I will take the approach to setting up the Apache SkyWalking as a *self-managed* APM tool within the Kubernetes cluster created in HPE GreenLake for Private Cloud Enterprise. This mainly takes into account the security concerns in HPE GreenLake product environment.
```
[the remark this answers is not in the document]
Comment (GuopingJia) [reply content/blog/set-up-apache-skywalking-for-k8s-and-vm-monitoring-in-hpe-greenlake-private-cloud.md:29]: You are absolutely right. HPE GreenLake provides excellent security. By setting up the APM tool within this environment, rather than outside, it can benefit from such security.

File: content/blog/set-up-apache-skywalking-for-k8s-and-vm-monitoring-in-hpe-greenlake-private-cloud.md (hunk 9 above)
Comment (GuopingJia) [reply content/blog/set-up-apache-skywalking-for-k8s-and-vm-monitoring-in-hpe-greenlake-private-cloud.md:60]: Have rephrased those "it". Need your review :-)

File: content/blog/set-up-apache-skywalking-for-k8s-and-vm-monitoring-in-hpe-greenlake-private-cloud.md [hunk 18]
```diff
@@ -0,0 +1,382 @@
...
+
+![](/img/java-app-map.png)
+
+### Deploy a multi-tier application
+
+A﻿s the second demo application, I will deploy a multi-tier *music* application, available as part of [Apache SkyWalking showcase application](https://github.com/apache/skywalking-showcase). This multi-tier music application consists of a frontend app server and its UI, backend gateway service, recommendation service and songs service, together with a *H2* database. Each microservice is implemented with different programming language, *NodeJS*, *React*, *Spring*, *Python*, etc.
+
+![](/img/multl-tier-app-music.png)
+
+I﻿n order to monitor this multi-tier *music* application from Apache SkyWalking, you need to pick up the SkyWalking agent per programming language and rebuild corresponding service to instrument the application to the SkyWalking OAP server.
```
[the remark this answers is not in the document]
Comment (GuopingJia) [reply content/blog/set-up-apache-skywalking-for-k8s-and-vm-monitoring-in-hpe-greenlake-private-cloud.md:265]: Have rephrased it. Please review.

File: content/blog/set-up-apache-skywalking-for-k8s-and-vm-monitoring-in-hpe-greenlake-private-cloud.md [hunk 19]
```diff
@@ -0,0 +1,382 @@
...
+    ├── gradle
+    ├── gradle.properties
+    ├── gradlew
+    ├── gradlew.bat
+    ├── Makefile
+    ├── settings.gradle
+    └── src
+```
+
+A﻿fter image files are rebuilt with the agents, the multi-tier *music* application can be deployed to the Kubernetes cluster:
```
[the remark this answers is not in the document]
Comment (GuopingJia) [reply content/blog/set-up-apache-skywalking-for-k8s-and-vm-monitoring-in-hpe-greenlake-private-cloud.md:308]: I think you are right. I will go with "in the Kubernetes cluster".

**Comment by GuopingJia** [comment]
> Thank you, Guoping! After this one, just one more for me to review.
> 
> All the best, Dale

Thanks Dale for your comments and good suggestions. 

The new patch has addressed all your comments. Let me know if there is still anything we want me to change. 

Guoping

File: content/blog/set-up-apache-skywalking-for-k8s-and-vm-monitoring-in-hpe-greenlake-private-cloud.md (hunk 17 above)
[the remark this answers is not in the document]
Comment (drensing) [reply content/blog/set-up-apache-skywalking-for-k8s-and-vm-monitoring-in-hpe-greenlake-private-cloud.md]: Okay, so it should be rewritten to express what you just said, that by setting up the APM tool within this environment, it can benefit from the security features of HPE GreenLake.

[the remark this answers is not in the document]
Comment (GuopingJia) [reply content/blog/set-up-apache-skywalking-for-k8s-and-vm-monitoring-in-hpe-greenlake-private-cloud.md]: Thanks Dale.  It's been addressed in the new patch.

File: content/blog/set-up-apache-skywalking-for-k8s-and-vm-monitoring-in-hpe-greenlake-private-cloud.md (hunk 8 above)
Comment (drensing) [reply content/blog/set-up-apache-skywalking-for-k8s-and-vm-monitoring-in-hpe-greenlake-private-cloud.md]: It looks good. Thanks!

File: content/blog/set-up-apache-skywalking-for-k8s-and-vm-monitoring-in-hpe-greenlake-private-cloud.md (hunk 9 above)
Comment (drensing) [reply content/blog/set-up-apache-skywalking-for-k8s-and-vm-monitoring-in-hpe-greenlake-private-cloud.md]: Yup! Rewrite works!

File: content/blog/set-up-apache-skywalking-for-k8s-and-vm-monitoring-in-hpe-greenlake-private-cloud.md (hunk 18 above)
[the remark this answers is not in the document]
Comment (drensing) [reply content/blog/set-up-apache-skywalking-for-k8s-and-vm-monitoring-in-hpe-greenlake-private-cloud.md]: Oh, yeah... that's much better. Much clearer. Thank you.

**Comment by drensing** [comment]
Re-reviewed the things you asked about. Looks great. I’m green. Thank you.

**Comment by GuopingJia** [comment]
The new patch has addressed the comment from Fred on privileged container policy.
